### PR TITLE
thorfinn: round11 — weight_decay=1e-3 (reg sweep, single delta)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

SOTA uses `weight_decay=5e-4` (default). The weight decay lever has **never been swept on tay** at the Lion lr=1e-4 + EMA=0.999 compound base. A 2× increase to `wd=1e-3` provides stronger L2 regularization, which could:

1. **Reduce the val→test generalization gap.** The SOTA val/test ratio is 1.115 (val 9.484 → test 10.580). Stronger weight decay should shrink this gap on a small dataset (400 train, 34 val, 50 test cases).
2. **Complement Lion's sign-gradient updates.** Lion's constant-magnitude steps may benefit from tighter weight norm control — weight decay acts as an implicit per-parameter scale regulation.

**Why 1e-3 and not 2e-3:** frieren's pod is stuck on PR #147 (wd=2e-3), giving no data. Starting at 1e-3 maps the midpoint between SOTA (5e-4) and the frieren target (2e-3). Single data point fills the gap.

**Risk:** Excessive regularization slows learning; trajectory ep1–2 will reveal if it's too aggressive.

## Instructions

Single change from SOTA: change `--weight-decay 5e-4` → `--weight-decay 1e-3`.

**Exact command:**
```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent thorfinn \
  --optimizer lion --lr 1e-4 --weight-decay 1e-3 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --use-ema --ema-decay 0.999 \
  --wandb-group tay-round11-wd-1e-3 \
  --wandb-name thorfinn/round11-wd-1e-3
```

This is the **only** change vs SOTA `d03oghpp`: `--weight-decay 1e-3` (SOTA uses `5e-4`).

Note: Do **not** add `--lr-cosine-t-max` — SOTA `d03oghpp` used `lr_cosine_t_max=0` (flat LR, no cosine).

## Baseline

Current SOTA — PR #115 (thorfinn, lr=1e-4 + EMA=0.999), test_abupt **10.580**, val 9.484 (ep9), W&B run `d03oghpp`.

| Metric | PR #115 SOTA | AB-UPT ref |
|---|---:|---:|
| `abupt` | **10.580** | — |
| `surface_pressure` | 5.690 | 3.82 |
| `wall_shear` | 10.419 | 7.29 |
| `volume_pressure` | 12.740 | 6.08 |
| `tau_x` | 8.908 | 5.35 |
| `tau_y` | 12.491 | 3.65 |
| `tau_z` | 13.071 | 3.63 |

**Reference val trajectory (SOTA):** 53.75 / 24.15 / 16.51 / 13.47 / 11.83 / 10.88 / 10.16 / 9.73 / 9.48

## Results

_(fill in after training completes)_

Post: best `val_primary/abupt_axis_mean_rel_l2_pct`, final `test_primary/abupt_axis_mean_rel_l2_pct`, per-component test breakdown, W&B run URL. Also note whether the val→test ratio improved vs SOTA (target: below 1.115).

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT thorfinn:

## Results

**TL;DR: wd=1e-3 regresses test_abupt to 11.058 (+4.5% vs SOTA 10.580). Hypothesis #1 (smaller val→test gap) marginally true but absolute test is much worse. wd=5e-4 is already at/below the optimum — the gradient in WD points down, not up.**

### Headline test metrics (best-val checkpoint reload)

| Metric | wd=1e-3 (this) | wd=5e-4 (SOTA `d03oghpp`) | Δ abs | Δ % | AB-UPT |
|---|---:|---:|---:|---:|---:|
| `abupt` | **11.058** | 10.580 | +0.478 | +4.5% | — |
| `surface_pressure` | 6.080 | 5.690 | +0.390 | +6.9% | 3.82 |
| `volume_pressure` | **12.722** | 12.740 | −0.018 | **−0.14%** | 6.08 |
| `wall_shear` | 11.019 | 10.419 | +0.600 | +5.8% | 7.29 |
| `tau_x` | 9.385 | 8.908 | +0.477 | +5.4% | 5.35 |
| `tau_y` | 13.274 | 12.491 | +0.783 | +6.3% | 3.65 |
| `tau_z` | 13.829 | 13.071 | +0.758 | +5.8% | 3.63 |

Best `val_primary/abupt`: **9.986** at ep9 (vs SOTA 9.484, +5.3% worse).
**val→test ratio: 1.107** (vs SOTA 1.115). Marginal improvement (−0.7% relative gap) — direction predicted by the hypothesis, magnitude does not compensate for the absolute regression.

### Val trajectory vs SOTA reference

| Ep | this (wd=1e-3) | SOTA (wd=5e-4) | Δ% |
|---:|---:|---:|---:|
| 1 | 59.38 | 53.75 | +10.5% |
| 2 | 26.42 | 24.15 | +9.4% |
| 3 | 18.06 | 16.51 | +9.4% |
| 4 | 14.46 | 13.47 | +7.4% |
| 5 | 12.57 | 11.83 | +6.3% |
| 6 | 11.47 | 10.88 | +5.4% |
| 7 | 10.77 | 10.16 | +6.0% |
| 8 | 10.23 | 9.73 | +5.1% |
| 9 | **9.99** | **9.48** | +5.3% |

The slowdown is largest at ep1 (+10.5%) and narrows but never closes. Curves are parallel after ep4 — wd=1e-3 is just a uniformly slower trajectory, not a different shape.

### Run details

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent thorfinn \
  --optimizer lion --lr 1e-4 --weight-decay 1e-3 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --use-ema --ema-decay 0.999 \
  --wandb-group tay-round11-wd-1e-3 \
  --wandb-name thorfinn/round11-wd-1e-3
```

- **W&B run (rank0):** `7rp28zrm` — group `tay-round11-wd-1e-3` — https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/7rp28zrm
- Runtime: **270.75 min** (9 val epochs, hit timeout — same budget as SOTA)
- Best epoch: **9** (epoch_time ≈ 1193 s = 19.9 min)
- Peak GPU memory: ~72% of 96GB ≈ 69.4 GB / GPU (8× RTX PRO 6000 Blackwell)
- DDP world_size=8, batch_size=4 (=32 effective)

### What happened

The hypothesis had two predicted effects:
1. **Smaller val→test gap** — *marginally true*. Ratio 1.107 vs 1.115 (Δ = −0.7% relative).
2. **Complement Lion's sign-gradient updates** — *false*. Lion on this 4L/512d/8h/400-cars setup is not under-regularized at wd=5e-4; doubling wd uniformly slowed the trajectory across all epochs by 5–10%.

The regularization premise is real but the operating point at wd=5e-4 is already at-or-below the optimum on this compound base. The slope of the gain function in WD is negative going from 5e-4 → 1e-3 on every metric except volume_pressure (which is essentially insensitive — the volume head is the laggard relative to AB-UPT and likely under-trained, so weight decay has little to fight).

**Implication for #147 (frieren wd=2e-3):** strongly expected to be even worse than this. Same direction, larger step → larger regression. The midpoint data fills the gap between SOTA and #147 with a clear monotonic regression.

### Note on duplicate run

A prior thorfinn invocation launched the experiment a second time (started 2026-05-01 11:04 UTC) while the original run had already completed at ~10:49 UTC with full test results. I killed the duplicate (PID 213255 + 8 ranks freed) so the GPUs are available for the next assignment. The 8 W&B runs from that aborted launch (group `tay-round11-wd-1e-3`, started at `2026-05-01T11:04:39Z`) will auto-mark as crashed once heartbeats time out.

### Suggested follow-ups

1. **Sweep DOWN, not up.** Since 5e-4 → 1e-3 regressed monotonically, the optimum may be at or below SOTA. Try wd=2.5e-4 or wd=1e-4 as a single-delta. Probability of winning is non-trivial: Lion's sign-only updates respond more sharply to per-param scale changes than AdamW, so a smaller wd with the lr=1e-4/EMA=0.999 stack could hit a sweet spot.
2. **Drop weight decay as a val→test gap lever.** 0.7% gap reduction at the cost of +4.5% absolute is a poor trade. If reducing the generalization gap matters, prefer dropout (currently 0.0) or stochastic depth — they don't directly bias the loss landscape the way wd does.
3. **Skip frieren #147 if it's still queued.** The 2e-3 point is predictable from this result + SOTA — both axes (val convergence rate, test absolute) will be worse. Free that pod for a more informative experiment.
4. **Volume_pressure insensitivity hint.** That `volume_pressure` is flat across wd ∈ {5e-4, 1e-3} while every surface metric regresses 5–7% suggests the volume head is bottlenecked by something other than weight scale (likely capacity or loss weighting). Worth a separate experiment with `--volume-loss-weight 2.0` re-added to the SOTA stack — the BASELINE.md note already flags this as expected to push abupt toward ~10.3.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #163 because ADVISOR: Closing PR #163 — wd=1e-3 clearly regresses all metrics vs SOTA wd=5e-4.

**Decision: Close (dead end in this direction)**

Your analysis is excellent and thorough. The results are unambiguous:
- test_abupt: 11.058 vs SOTA 10.580 (+4.5% worse)
- All surface metrics regressed 5–7%
- The val trajectory shows a uniformly parallel shift — same curve shape, just slower

The parallel curves after ep4 is the key diagnostic: this isn't a different optimizer trajectory, just a scaled-down version. wd at 5e-4 is already at or below the optimum for this Lion/lr=1e-4/4L/512d stack.

**Your suggestion #1 is the right call:** sweep DOWN to wd=2.5e-4 or wd=1e-4. The gradient in WD points downward from 5e-4, so the optimum is likely below it. I'll assign this to another student.

**On frieren #147 (wd=2e-3):** Agreed — that result is predictable from yours. I'll close it once I confirm it hasn't produced anything unexpected, but your data makes the outcome clear.

**On volume_pressure insensitivity:** Good catch. You've correctly identified that volume_pressure is not regularization-sensitive in this regime — it's capacity-limited. The vol_loss_weight experiment (PR #187, nezuko) is running now.

Thank you for the detailed write-up and the self-cleanup of the duplicate run. Well done..
